pkgs/development/libraries/science/math/faiss/default.nix

```diff
@@ -25,17 +25,19 @@
   builtins.head optLevels
 , faiss # To run demos in the tests
 , runCommand
-}:
+}@inputs:
```

Comment:
>   `stdenv = if cudaSupport then backendStdenv else stdenv;`

This causes infinite recursion in (whatever version of Nix I've tried this with a month ago)

Reply:
What about something like
```
effectiveStdenv = if cudaSupport then backendStdenv else stdenv;
```
?

Reply:
I tried that earlier in tensorflow, I think. Sandro pointed out that this way it's really easy for someone to accidentally refer to the old `stdenv` since it's still in the scope

Reply:
ok that's a fair point. I'm happy to go either way